examples/wasi_args_test.go

```diff
@@ -0,0 +1,38 @@
+package examples
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/wazero/wasi"
+	"github.com/tetratelabs/wazero/wasm"
+	"github.com/tetratelabs/wazero/wasm/binary"
+	"github.com/tetratelabs/wazero/wasm/interpreter"
+)
+
+func Test_WASI_args(t *testing.T) {
+	buf, err := os.ReadFile("testdata/wasi_args.wasm")
+	require.NoError(t, err)
+
+	mod, err := binary.DecodeModule(buf)
+	require.NoError(t, err)
+
+	store := wasm.NewStore(interpreter.NewEngine())
+	require.NoError(t, err)
+
+	args, err := wasi.Args([]string{"foo", "bar", "foobar", "", "baz"})
+	require.NoError(t, err)
+	wasiEnv := wasi.NewEnvironment(args, wasi.Stdout(os.Stderr))
+
+	err = wasiEnv.Register(store)
+	require.NoError(t, err)
+
+	err = store.Instantiate(mod, "test")
+	require.NoError(t, err)
+
+	// Let TinyGo runtime initialize the WASI environment by calling main.
```

Comment:
this feels like a test that should be in tinygo.. what makes me curious is why we have this and not for example a %.wat file importing the expected functions, then using testify here to validate they match wasi.args

Reply:
ex. to be concrete, you can export and imported function.. that seems like the easiest path to check wasi arg functions are returning the correct stuff. It feels like the tinygo runtime sanity tests might be better put into tinygo where they can maintain it.